### PR TITLE
Oxpecker support

### PR DESCRIPTION
**Pull request by xperiandri**

### CODE REVIEW

File: .github/workflows/publish_ci.yml [hunk 1]
```diff
@@ -31,7 +31,7 @@ jobs:
 
     - name: Add version to global.json
       run: |
-        $version = "7.0.406"
+        $version = "8.0.202"
```
Comment (valbers) [inline .github/workflows/publish_ci.yml]: Maybe in the future do this sort of change in a separate PR?

File: src/FSharp.Data.GraphQL.Server.Giraffe/FSharp.Data.GraphQL.Server.Giraffe.fsproj [hunk 2]
```diff
@@ -0,0 +1,34 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <TargetFrameworks>$(PackageTargetFrameworks)</TargetFrameworks>
+    <GenerateDocumentationFile>true</GenerateDocumentationFile>
+    <Deterministic>true</Deterministic>
+    <Description>FSharp implementation of Facebook GraphQL query language (Application Infrastructure)</Description>
```
Comment (valbers) [inline src/FSharp.Data.GraphQL.Server.Giraffe/FSharp.Data.GraphQL.Server.Giraffe.fsproj]:
```suggestion
    <Description>FSharp implementation of Facebook GraphQL query language (Giraffe integration)</Description>
```

File: src/FSharp.Data.GraphQL.Server.Oxpecker/FSharp.Data.GraphQL.Server.Oxpecker.fsproj [hunk 3]
```diff
@@ -0,0 +1,36 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <!--<TargetFrameworks>$(PackageTargetFrameworks)</TargetFrameworks>-->
+    <TargetFramework>net8.0</TargetFramework>
+    <GenerateDocumentationFile>true</GenerateDocumentationFile>
+    <Deterministic>true</Deterministic>
+    <Description>FSharp implementation of Facebook GraphQL query language (Application Infrastructure)</Description>
```
Comment (valbers) [inline src/FSharp.Data.GraphQL.Server.Oxpecker/FSharp.Data.GraphQL.Server.Oxpecker.fsproj]:
```suggestion
    <Description>FSharp implementation of Facebook GraphQL query language (Oxpecker integration)</Description>
```

File: src/FSharp.Data.GraphQL.Server.Oxpecker/HttpEndpoints.fs
Comment (valbers) [inline src/FSharp.Data.GraphQL.Server.Oxpecker/HttpEndpoints.fs:1]: This implementation is almost the same as for Giraffe. Maybe we could think of a way of sharing the same code between both integrations.
For example: maybe we can define our own `HttpHandler` (in Giraffe terms) delegate (or an interface) which is to be implemented by both Giraffe and Oxpecker integrations.

File: src/FSharp.Data.GraphQL.Server/Ast.fs [hunk 4]
```diff
@@ -1,8 +1,10 @@
-module FSharp.Data.GraphQL.Server.AspNetCore.Ast
+module FSharp.Data.GraphQL.Server.Ast
 
 open System.Collections.Immutable
 open FSharp.Data.GraphQL
 
+/// A list of fields that are reserved for the GraphQL introspection system.
+[<CompiledName "MetaTypeFields">]
```
Comment (valbers) [inline src/FSharp.Data.GraphQL.Server/Ast.fs:7]: What's the motivation for using a different compiled name?

**Review by valbers — CHANGES_REQUESTED** [review]
Please see my comments.

File: .github/workflows/publish_ci.yml (hunk 1 above)
Comment (xperiandri) [reply .github/workflows/publish_ci.yml]: Yep, the issue is that I was not managed to make .NET buildable with SDK 8

File: src/FSharp.Data.GraphQL.Server/Ast.fs (hunk 4 above)
Comment (xperiandri) [reply src/FSharp.Data.GraphQL.Server/Ast.fs:7]: Make it friendly to C# as it is a public API

File: src/FSharp.Data.GraphQL.Server.Oxpecker/HttpEndpoints.fs
Comment (xperiandri) [reply src/FSharp.Data.GraphQL.Server.Oxpecker/HttpEndpoints.fs:1]: Good point, I'll think how to make that

Comment (xperiandri) [reply src/FSharp.Data.GraphQL.Server.Oxpecker/HttpEndpoints.fs:1]: I've addressed your suggestion

**Comment by xperiandri** [comment]
@valbers, PR is ready for review
@njlr please review too

File: src/FSharp.Data.GraphQL.Server.AspNetCore/GraphQLRequest.fs [hunk 5]
```diff
@@ -0,0 +1,252 @@
...
+open Microsoft.Extensions.Logging
+open Microsoft.Extensions.Options
+
+open FSharp.Data.GraphQL
+open FsToolkit.ErrorHandling
+
+open FSharp.Data.GraphQL.Server
+
+/// Provides logic to parse and execute GraphQL request
+type GraphQLRequest<'Root> (
```
Comment (valbers) [inline src/FSharp.Data.GraphQL.Server.AspNetCore/GraphQLRequest.fs]: `GraphQLRequest` as a name sounds too "static" and passive, like an HTTP request, which is an object containing information about a request. How about `GraphQLRequestHandler` or `GraphQLRequestHandlingLogic`?

**Review by valbers — DISMISSED** [review]
Please consider my remark about the naming of the class.

File: src/FSharp.Data.GraphQL.Server.AspNetCore/GraphQLRequest.fs (hunk 5 above)
Comment (xperiandri) [reply src/FSharp.Data.GraphQL.Server.AspNetCore/GraphQLRequest.fs]: Renamed

**Review by Juriyx — APPROVED** [review]

**Pull request merged**